### PR TITLE
[SLING-9417] Versioning support for nt:file as child of versioned node

### PR DESCRIPTION
**Pull request by alldaysunshine**
See issue in Jira.

### CODE REVIEW

File: src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import javax.jcr.Property;
 import javax.jcr.RepositoryException;
 import javax.jcr.Value;
+import org.apache.jackrabbit.JcrConstants;
```
Comment (kwin) [inline src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java:36]: Please use the constants from https://docs.adobe.com/docs/en/spec/javax.jcr/javadocs/jcr-2.0/constant-values.html instead and remover the static imports for org.apache.jackrabbit.JcrConstants as well.

File: src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResourceMetadata.java [hunk 2]
```diff
@@ -35,6 +35,8 @@
 import javax.jcr.Property;
 import javax.jcr.RepositoryException;
 import javax.jcr.ValueFormatException;
+import org.apache.jackrabbit.JcrConstants;
+import static org.apache.jackrabbit.JcrConstants.NT_FILE;
```
Comment (kwin) [inline src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResourceMetadata.java:39]: Please clean up imports and only use constants from https://docs.adobe.com/docs/en/spec/javax.jcr/javadocs/jcr-2.0/constant-values.html

File: src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java [hunk 3]
```diff
@@ -182,7 +183,10 @@ private InputStream getInputStream() {
             try {
                 // find the content node: for nt:file it is jcr:content
                 // otherwise it is the node of this resource
-                Node content = node.isNodeType(NT_FILE)
+                Node content = (node.isNodeType(NT_FILE) ||
+                                (node.isNodeType(JcrConstants.NT_FROZENNODE) &&
+                                 node.hasProperty(JcrConstants.JCR_FROZENPRIMARYTYPE) &&
```
Comment (rombert) [inline src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java:188]: I think this is not necessary. `jcr:frozenPrimaryType` is mandatory and autocreated for `nt:frozenNode` https://docs.adobe.com/docs/en/spec/jcr/2.0/3_Repository_Model.html#3.13.4.1%20nt:frozenNode

**Comment by alldaysunshine** [comment]
Thank you for the review and suggestions; these have been implemented. The original code also uses org.apache.jackrabbit.JcrConstants, so I thought this would be fine.

**Comment by rombert** [comment]
@kwin , does this look ok to you now?

**Comment by kwin** [comment]
Yes, although I would like to move all constants to the javax.jcr packages. But this is independent of this fix.

**Comment by rombert** [comment]
@alldaysunshine - can you please:

- squash the two commits into a single one
- reference the Jira issue in the commit title

? Then I can merge this

**Comment by alldaysunshine** [comment]
I squashed the two commits into a single one by rebasing. Hope this was done in the right way and it can be merged.

**Comment by rombert** [comment]
Perfect, thanks a lot for the contribution @alldaysunshine !

**Pull request merged**